hknweb/events/views.py

```diff
@@ -4,13 +4,32 @@
 from django.contrib import messages
 from django.shortcuts import get_object_or_404
 from django.views.decorators.csrf import csrf_exempt #doing this for now bc idk how to make csrf work
-from django.db.models import F # Avoids changing database values without risking a race condition
+from django.conf import settings
 from django.contrib.auth.decorators import login_required, permission_required
+import datetime
+import pytz
 
 from .models import Event, Rsvp
 from .forms import EventForm
 from hknweb.models import Profile
 
+# decorators
+
+# used for things only officers and candidates can access
+def check_account_access(func):
+    def check_then_call(request, *args, **kwargs):
+        if not is_cand_or_officer(request.user):
+            return render(request, "errors/401.html", status=401)
+        return func(request, *args, **kwargs)
+    return check_then_call
+
+def is_cand_or_officer(user):
+    return user.groups.filter(name=settings.CAND_GROUP).exists() or \
+           user.groups.filter(name=settings.OFFICER_GROUP).exists()
```

Comment:
See below comment on suggestion to switch to fine-grained permission access control over coarse group access control.